### ISSUE TITLE
macOS 12 runner image will be removed by 12/3/2024

### ISSUE THREAD

**Issue by gaul** [s3fs-fuse/s3fs-fuse#2552]
The macOS 12 runner image will be removed by December 3rd, 2024. To raise awareness of the upcoming removal, jobs using macOS 12 will temporarily fail during scheduled time periods defined below:

* November 4, 9:00 AM - 7:00 PM EST
* November 11, 9:00 AM - 7:00 PM EST
* November 18, 9:00 AM - 7:00 PM EST
* November 25, 9:00 AM - 7:00 PM EST

What you need to do

Jobs using the macos-12 YAML workflow label should be updated to macos-15, macos-14, macos-13, or macos-latest. You can always get up-to-date information on our tools by reading about the software in the [runner images](https://github.com/actions/runner-images?elqTrackId=52d12286b0ab49208138b8f06e3af7e9&elq=d9ca020bc2074e2c800b4ada53798ae8&elqaid=4233&elqat=1&elqCampaignId=4442&elqak=8AF539AA690C65F019049E0E1A8B640381522B665242355DEA6735AE30E1F2538DEE) repository. Please contact GitHub [Support](https://support.github.com/contact?form%5Bsubject%5D=Re%3A+GitHub+Actions&tags=dotcom-contact-params&elqTrackId=7a78e8710a0c4d378d4de10f9828752e&elq=d9ca020bc2074e2c800b4ada53798ae8&elqaid=4233&elqat=1&elqCampaignId=4442&elqak=8AF592F31F339E082BFF97F54F17F450F2B02B665242355DEA6735AE30E1F2538DEE) if you run into any problems or need help.

**Comment by ggtakec** [comment]
I noticed this notice too.
After 1.95 is released, we should try 13->14->15.

**Comment by ggtakec** [comment]
@gaul 
In the past few days (I think yesterday), there has been a change in the GHA Runner for macos-13, and an error has started to occur.

```
configure: error: Package requirements (fuse-t >= 1.0.20 libcurl >= 7.0 libxml-2.0 >= 2.6 libcrypto >= 0.9 ) were not met:

Package 'ldap', required by 'libcurl', not found

Consider adjusting the PKG_CONFIG_PATH environment variable if you
installed software in a non-standard prefix.

Alternatively, you may set the environment variables DEPS_CFLAGS
and DEPS_LIBS to avoid the need to call pkg-config.
See the pkg-config man page for more details.
```
It seems to depend on pkg-config for curl or ldap.

We'll have to wait and see.
I'll continue to follow this issue (due to the ongoing PR).

**Comment by ggtakec** [comment]
@gaul
The problem with dependencies on macOS 13 seems to have been resolved. (Currently, there are no errors when running CI.)
(I may be wrong) I think the dependency was resolved on the curl side.

https://github.com/curl/curl/pull/15573#issuecomment-2476813967

**Comment by gaul** [comment]
This seems fixed since #2619 passed.

**Comment by ggtakec** [comment]
@gaul 
I also tried testing on macOS 14/15, and found that the fuse related header files were missing.
(This can be avoided by installing macfuse(brew package) and adding the include path(/usr/loca/include/fuse) when compiling.)
And, just like before, I got an error saying that stdbuf could not be found.

For the time being, I will only run CI on macos13.
